CHANGELOG.md

```diff
@@ -1,331 +1,1350 @@
 
-* Reducing the likelihood that the connection manager will kill connections to
-  peers in the routing table.
-* Keeping peers in the routing table, even if we get disconnected from them.
-* Actively and frequently querying the DHT to keep our routing table full.
+- Reducing the likelihood that the connection manager will kill connections to peers in the routing table.
+- Keeping peers in the routing table, even if we get disconnected from them.
+- Actively and frequently querying the DHT to keep our routing table full.
+- Prioritizing useful peers that respond to queries quickly.
 
 ##### Testing
 
-The DHT rewrite was made possible by our new testing framework,
-[testground](https://github.com/ipfs/testground), which allows us to spin up
-multi-thousand node tests with simulated real-world network conditions. With
-testground and some custom analysis tools, we were able to gain confidence that
-the new DHT implementation behaves correctly.
+The DHT rewrite was made possible by [Testground](https://github.com/ipfs/testground/), our new testing framework. Testground allows us to spin up multi-thousand node tests with simulated real-world network conditions. By combining Testground and some custom analysis tools, we were able to gain confidence that the new DHT implementation behaves correctly.
+
+##### Provider Record Changes
+
+When you add content to your IPFS node, you advertise this content to the network by announcing it in the DHT. We call this "providing".
+
+However, go-ipfs has multiple ways to address the same underlying bytes. Specifically, we address content by content ID (CID) and the same underlying bytes can be addressed using (a) two different versions of CIDs (CIDv1 and CIDv2) and (b) with different "codecs" depending on how we're interpreting the data.
+
+Prior to go-ipfs 0.5.0, we used the content id (CID) in the DHT when sending out provider records for content. Unfortunately, this meant that users trying to find data announced using one CID wouldn't find nodes providing the content under a different CID.
 
-#### Refactored Bitswap
+In go-ipfs 0.5.0, we're announcing data by _multihash_, not _CID_. This way, regardless of the CID version used by the peer adding the content, the peer trying to download the content should still be able to find it.
 
-This release includes a _major_ [bitswap refactor][bitswap-refactor] running a
-new, but backwards compatible, bitswap protocol. We expect these changes to
-improve performance significantly.
+**Warning:** as part of the network, this could impact finding content added with CIDv1. Because go-ipfs 0.5.0 will announce and search for content using the bare multihash (equivalent to the v0 CID), go-ipfs 0.5.0 will be unable to find CIDv1 content published by nodes prior to go-ipfs 0.5.0 and vice-versa. As CIDv1 is _not_ enabled by default so we believe this will have minimal impact. However, users are _strongly_ encouraged to upgrade as soon as possible.
```

Comment:
```suggestion
**Warning:** as part of the network, this could impact finding content added with CIDv1. Because go-ipfs 0.5.0 will announce and search for content using the bare multihash (equivalent to the v0 CID), go-ipfs 0.5.0 will be unable to find CIDv1 content published by nodes prior to go-ipfs 0.5.0 and vice-versa. As CIDv1 is _not_ enabled by default we believe this will have minimal impact. However, users are _strongly_ encouraged to upgrade as soon as possible.
```